sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -118,97 +132,78 @@ class ReadMessagesFromPubSub(PTransform):
-        that no duplicate data will be delivered on the Pub/Sub stream. In this
-        case, deduplication of the stream will be strictly best effort.
     """
     super(ReadMessagesFromPubSub, self).__init__()
     self.topic = topic
     self.subscription = subscription
     self.id_label = id_label
+    self.timestamp_attribute = timestamp_attribute
 
   def get_windowing(self, unused_inputs):
     return core.Windowing(window.GlobalWindows())
 
   def expand(self, pcoll):
     p = (pcoll.pipeline
          | _ReadFromPubSub(self.topic, self.subscription, self.id_label,
-                           with_attributes=True))
+                           with_attributes=True,
+                           timestamp_attribute=self.timestamp_attribute))
     return p
 
 
-class ReadStringsFromPubSub(PTransform):
-  """A ``PTransform`` for reading utf-8 string payloads from Cloud Pub/Sub.
+class ReadPayloadsFromPubSub(PTransform):
+  """A ``PTransform`` for reading raw payloads from Cloud Pub/Sub.
 
-  Outputs elements of type ``unicode``, decoded from UTF-8.
+  Outputs elements of type ``str``.
   """
 
-  def __init__(self, topic=None, subscription=None, id_label=None):
-    """Initializes ``ReadStringsFromPubSub``.
-
-    Args:
-      topic: Cloud Pub/Sub topic in the form "projects/<project>/topics/
-        <topic>". If provided, subscription must be None.
-      subscription: Existing Cloud Pub/Sub subscription to use in the
-        form "projects/<project>/subscriptions/<subscription>". If not
-        specified, a temporary subscription will be created from the specified
-        topic. If provided, topic must be None.
-      id_label: The attribute on incoming Pub/Sub messages to use as a unique
-        record identifier.  When specified, the value of this attribute (which
-        can be any string that uniquely identifies the record) will be used for
-        deduplication of messages.  If not provided, we cannot guarantee
-        that no duplicate data will be delivered on the Pub/Sub stream. In this
-        case, deduplication of the stream will be strictly best effort.
-    """
-    super(ReadStringsFromPubSub, self).__init__()
+  def __init__(self, topic=None, subscription=None, id_label=None,
+               timestamp_attribute=None):
+    super(ReadPayloadsFromPubSub, self).__init__()
     self.topic = topic
     self.subscription = subscription
     self.id_label = id_label
+    self.timestamp_attribute = timestamp_attribute
 
   def get_windowing(self, unused_inputs):
     return core.Windowing(window.GlobalWindows())
 
   def expand(self, pcoll):
     p = (pcoll.pipeline
          | _ReadFromPubSub(self.topic, self.subscription, self.id_label,
-                           with_attributes=False)
+                           with_attributes=False,
+                           timestamp_attribute=self.timestamp_attribute))
+    return p
+
+
+class ReadStringsFromPubSub(ReadPayloadsFromPubSub):
```

Comment:
Do we actually need this transform? How is this handled in Java SDK?

Reply:
Please see my comment above.